### PR TITLE
Get messages endpoint implemented

### PR DESCRIPTION
**Pull request by RyanM-Dev**

### CODE REVIEW

File: api/chat.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import (
+	"log"
+
+	"github.com/gin-gonic/gin"
+	"github.com/goccy/go-json"
+	"github.com/mhghw/fara-message/db"
+)
+
+func NewChatAPI(c *gin.Context) {
```
Comment (mhghw) [inline api/chat.go]: why does your function name has an API at the end?

File: api/chat.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	var requestBody NewChat
```
Comment (mhghw) [inline api/chat.go]: NewChat is a function name, try using a better name like NewChatRequest

File: api/chat.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	err := c.BindJSON(&requestBody)
+	if err != nil {
+		log.Print("failed to bind json", err)
```
Comment (mhghw) [inline api/chat.go]: you did not format it well, add a `,` and a space to the string

File: api/chat.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+		return
+	}
+	if err := db.NewChat(requestBody.ChatName, requestBody.ChatType, requestBody.Users); err != nil {
```
Comment (mhghw) [inline api/chat.go]: you also need to separate endpoints for chat types, client shouldn't specify the chat type for you, also direct messages has no chatName

File: api/chat.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		log.Print("failed to create chat", err)
```
Comment (mhghw) [inline api/chat.go]: bad formatting

File: api/chat.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	}
+	chatID := requestBody.ID
+	messages, err := db.GetChatMessages(int64(chatID))
+	if err != nil {
+		log.Print(err)
+		return
+	}
+	chatMessages, err := json.Marshal(messages)
+	if err != nil {
+		log.Print("failed to marshal json", err)
```
Comment (mhghw) [inline api/chat.go:55]: bad formatting

File: api/chat.go [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+	var requestBody Chat
+	err := c.BindJSON(&requestBody)
+	if err != nil {
+		log.Print("failed to bind json", err)
+		return
+	}
+	chatID := requestBody.ID
+	messages, err := db.GetChatMessages(int64(chatID))
+	if err != nil {
+		log.Print(err)
```
Comment (mhghw) [inline api/chat.go:50]: add some context to error

File: api/chat.go [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+		log.Print("failed to create chat", err)
+		return
+	}
+}
+
+func GetChatMessagesAPI(c *gin.Context) {
+	var requestBody Chat
+	err := c.BindJSON(&requestBody)
+	if err != nil {
+		log.Print("failed to bind json", err)
```
Comment (mhghw) [inline api/chat.go]: bad formatting

File: api/models.go [hunk 9]
```diff
@@ -1,10 +1,22 @@
...
 }
 
-type HTTPError struct{
+type HTTPError struct {
 	Message string `json:message`
 }
+
+type NewChat struct {
+	ChatName string    `json:"chatName"`
+	ChatType string    `json:"chatType"`
```
Comment (mhghw) [inline api/models.go]: you should not get chat name from client

File: db/models.go [hunk 10]
```diff
@@ -0,0 +1,20 @@
+package db
+
+import "time"
+
+type Chat struct {
+	ID          int64 `gorm:"primary_key"`
+	ChatName    string
+	CreatedTime time.Time
+	DeletedTime time.Time
+	ChatType    string
```
Comment (mhghw) [inline db/models.go]: use enum for ChatType
also it could be Type, and not ChatType

**Review by mhghw — CHANGES_REQUESTED** [review]

File: api/chat.go (hunk 4 above)
Comment (RyanM-Dev) [reply api/chat.go]: we can handle it in NEWCHATAPI

File: api/chat.go [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+import (
+	"log"
+
+	"github.com/gin-gonic/gin"
+	"github.com/goccy/go-json"
+	"github.com/mhghw/fara-message/db"
+)
+
+func NewDirectChatRequest(c *gin.Context) {
+	var requestBody NewDirectChat
```
Comment (mhghw) [inline api/chat.go]: function name should a NewDirectChat
strcut name should be NewDirectChatRequest

File: api/chat.go [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+	err := c.BindJSON(&requestBody)
+	if err != nil {
+		log.Print("failed to bind json, ", err)
```
Comment (mhghw) [inline api/chat.go:16]: use log.Printf

File: api/chat.go [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+		return
```
Comment (mhghw) [inline api/chat.go:17]: set http status code, here it could be 400

File: api/chat.go [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	if err := db.NewChat("", db.Direct, requestBody.Users); err != nil {
+		log.Print("failed to create chat, ", err)
+		return
```
Comment (mhghw) [inline api/chat.go:22]: use log.Printf
set status code

File: api/chat.go [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+	}
+}
+
+func NewGroupChatRequest(c *gin.Context) {
+	var requestBody NewGroupChat
```
Comment (mhghw) [inline api/chat.go]: same as direct

File: api/chat.go [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+	err := c.BindJSON(&requestBody)
+	if err != nil {
+		log.Print("failed to bind json, ", err)
```
Comment (mhghw) [inline api/chat.go:30]: use log.Printf

File: api/chat.go [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+		return
```
Comment (mhghw) [inline api/chat.go:31]: set status code

File: api/chat.go [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	if err := db.NewChat(requestBody.ChatName, db.Group, requestBody.Users); err != nil {
+		log.Printf("failed to create chat: %v", err)
+		return
```
Comment (mhghw) [inline api/chat.go:36]: use log.Printf
set status code

File: api/chat.go [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+	}
+}
+
+func GetChatMessagesAPI(c *gin.Context) {
```
Comment (mhghw) [inline api/chat.go]: API?

File: api/chat.go [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+	var requestBody Chat
+	err := c.BindJSON(&requestBody)
+	if err != nil {
+		log.Printf("failed to bind json: %v", err)
+		return
```
Comment (mhghw) [inline api/chat.go:45]: log.Printf
set status code

File: api/chat.go [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+	}
+	chatID := requestBody.ID
+	messages, err := db.GetChatMessages(int64(chatID))
```
Comment (mhghw) [inline api/chat.go:48]: why just not pass requestBody.ID to GetChatMessages?

File: api/chat.go [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+	if err != nil {
+		log.Print(err)
+		return
```
Comment (mhghw) [inline api/chat.go:51]: add some context with log.Printf
set status code

File: api/chat.go [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+	}
+	chatMessages, err := json.Marshal(messages)
+	if err != nil {
+		log.Print("failed to marshal json", err)
+		return
```
Comment (mhghw) [inline api/chat.go:56]: use log.Printf and set status code

File: api/models.go [hunk 24]
```diff
@@ -1,10 +1,24 @@
...
+	ChatName string    `json:"chatName"`
+	Users    []db.User `json:"users"`
+}
+type NewDirectChat struct {
+	Users []db.User `json:"users"`
+}
+
+type Chat struct {
+	ID int
+}
```
Comment (mhghw) [inline api/models.go:24]: Whats chat?
try adding more context to struct name
like GetChatRequest or GetChatBody

File: api/webserver.go [hunk 25]
```diff
@@ -11,7 +11,9 @@ func RunWebServer(port int) error {
 	router := gin.New()
 	router.POST("/user/register", Register)
 	router.Use(AuthMiddleware)
-	router.POST("/user/newchat", NewChatAPI)
+	router.POST("/new_direct_chat", NewDirectChatRequest)
+	router.POST("/new_Group_chat", NewGroupChatRequest)
+	router.GET("/chat/:id/messages", GetChatMessagesAPI)
```
Comment (mhghw) [inline api/webserver.go]: Fix here or change the handler

**Review by mhghw — CHANGES_REQUESTED** [review]

File: api/chat.go (hunk 15 above)
Comment (RyanM-Dev) [reply api/chat.go]: what do you mean?

File: api/chat.go (hunk 21 above)
Comment (RyanM-Dev) [reply api/chat.go:48]: i really don't know why i did that :D

File: api/webserver.go (hunk 25 above)
Comment (RyanM-Dev) [reply api/webserver.go]: whats the problem?

File: api/models.go (hunk 24 above)
Comment (RyanM-Dev) [reply api/models.go:24]: i assume it was temp for checking...

**Pull request merged**